File: packages/ui_login/tests/metadata.test.ts
import { describe, it, expect } from 'vitest'
import metadata from '../seed/metadata.json'

describe('UI Login Metadata', () => {
  it('should have valid package structure', () => {
    expect(metadata.packageId).toBe('ui_login')
    expect(metadata.name).toBe('Login Page')
    expect(metadata.version).toMatch(/^\d+\.\d+\.\d+$/)
  })

  it('should depend on ui_permissions', () => {
    expect(metadata.dependencies).toContain('ui_permissions')
  })

  it('should export login page', () => {
    expect(metadata.exports.pages).toContain('login')
  })
})

File: packages/archive/ui_level5/seed/scripts/tests/metadata.test.lua
-- Metadata validation tests for ui_level5 package
-- Uses lua_test framework

describe("Level 5 God Panel Package Metadata", function()
  local metadata = load_cases("metadata.json")
  
  it("should have valid package structure", function()
    expect(metadata.packageId).toBe("ui_level5")
    expect(metadata.name).toBe("Level 5 - God Panel")
    expect(metadata.version).toBeTruthy()
    expect(metadata.description).toBeTruthy()
  end)
  
  it("should have correct package ID format", function()
    expect(metadata.packageId).toMatch("^[a-z_0-9]+$")
  end)
  
  it("should have semantic version", function()
    expect(metadata.version).toMatch("^%d+%.%d+%.%d+$")
  end)
  
  it("should have exports defined", function()
    expect(metadata.exports).toBeTruthy()
  end)
  
  it("should have dependencies array", function()
    expect(metadata.dependencies).toBeType("table")
  end)
  
  it("should depend on ui_permissions", function()
    local found = false
    for _, dep in ipairs(metadata.dependencies) do
      if dep == "ui_permissions" then found = true break end
    end
    expect(found).toBe(true)
  end)
  
  it("should require minLevel 5", function()
    expect(metadata.minLevel).toBe(5)
  end)
  
  it("should export pages", function()
    expect(metadata.exports.pages).toBeType("table")
  end)
end)
